src/NLPModels.jl

```diff
@@ -272,6 +281,25 @@ Evaluate ``∇c(x)v``, the Jacobian-vector product at `x` in place.
 jprod!(::AbstractNLPModel, ::AbstractVector, ::AbstractVector, ::AbstractVector) =
   throw(NotImplementedError("jprod!"))
 
+"""
+    Jv = jprod!(nlp, rows, cols, vals, v, Jv)
+
+Evaluate ``∇c(x)v``, the Jacobian-vector product, where the Jacobian is given by
```

Comment:
Technically, ∇c(x) is the gradient matrix of the constraints, which is the transposed Jacobian. So there should be a transposed here, but it might be clearer to just write J(x).

Reply:
This appears in other places, so I created #245 to fix them all together.